tests/puppeteer/pages/BO/modules/moduleManager/index.js

```diff
@@ -38,4 +44,50 @@ module.exports = class moduleManager extends BOBasePage {
   async goToConfigurationPage(moduleName) {
     await this.page.click(this.configureModuleButton.replace('%MODULENAME', moduleName));
   }
+
+  /**
+   * Filter modules by status
+   * @param enabled
+   * @return {Promise<void>}
+   */
+  async filterByStatus(enabled) {
+    await Promise.all([
+      this.page.click(this.statusDropdownDiv),
+      this.page.waitForSelector(`${this.statusDropdownDiv}[aria-expanded='true']`),
+    ]);
+    await Promise.all([
+      this.page.click(this.statusDropdownItemLink.replace('%REF', enabled ? 1 : 0)),
+      this.page.waitForSelector(`${this.statusDropdownDiv}[aria-expanded='false']`),
+    ]);
+  }
+
+  /**
+   * Get status of module (enable/disable)
+   * @param moduleName
+   * @return {Promise<boolean|true>}
+   */
+  async isModuleEnabled(moduleName) {
+    return this.elementNotVisible(this.disableModuleButton.replace('%MODULENAME', moduleName), 1000);
+  }
+
+  /**
+   * Get all modules status
+   * @return {Promise<void>}
+   */
+  async getAllModulesStatus() {
+    const modulesStatus = [];
+    const allModulesNames = await this.getAllModulesNames();
+    for (let i = 0; i < allModulesNames.length; i++) {
+      const moduleStatus = await this.isModuleEnabled();
+      await modulesStatus.push({name: allModulesNames[i], status: moduleStatus});
+    }
+    return modulesStatus;
+  }
+
+  async getAllModulesNames() {
+    return this.page.$$eval(
```

Comment:
What is `$$eval` ?

Reply:
A puppeteer command to let us execute Javascript code in the page itself. It's really useful to target sepcific stuff or trigger events (like clicks) when Puppeteer is not behaving correctly.
We use it in a lot of places in the tests actually :)

Reply:
for more information https://pptr.dev/#?product=Puppeteer&version=v1.9.0&show=api-pageevalselector-pagefunction-args-1

Reply:
Top. Thanks